### PR TITLE
feat: Assignment 21 Exploring NuGet Package

### PR DESCRIPTION
**Pull request by Subhasri-V-P**
**Background: **
To use, create and manage NuGet Package.

**Implementation: ** 
1. A class library is created.
2. It is established as a NuGet package.
3. A console application is created to use the package created in task 1.
4. The console application generates Fibonacci series using MathNet.Numerics NuGet package.
5. The package is then published in BaGet server.

### CODE REVIEW

File: src/MyConsoleApplication/FibonacciGenerator.cs [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        /// CheckIfEmpty - checks if the list is empty.
+        /// </summary>
+        /// <param name="fibonacciSeries">Fibonacci list to be checked.</param>
+        public static void CheckIfEmpty(IEnumerable<BigInteger> fibonacciSeries)
+        {
+            if (!fibonacciSeries.Any())
+            {
+                Console.ForegroundColor = ConsoleColor.Yellow;
+                Console.WriteLine("Empty List returned!! Try inputting number greater than 0!!");
+                Console.ForegroundColor = ConsoleColor.White;
```
Comment (Shaalini-Subrammanian) [inline src/MyConsoleApplication/FibonacciGenerator.cs:55]:
```suggestion
                Console.ResetColor();
```

File: src/MyConsoleApplication/FibonacciGenerator.cs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        public static void DisplayList(IEnumerable<BigInteger> fibonacciSeries)
+        {
+            CheckIfEmpty(fibonacciSeries);
+            Console.ForegroundColor = ConsoleColor.Green;
+            foreach (BigInteger numbers in fibonacciSeries)
+            {
+                Console.Write($"{numbers} ");
+            }
+
+            Console.ForegroundColor = ConsoleColor.White;
```
Comment (Shaalini-Subrammanian) [inline src/MyConsoleApplication/FibonacciGenerator.cs:42]:
```suggestion
            Console.ResetColor();
```

**Review by Shaalini-Subrammanian — APPROVED** [review]

**Comment by Shaalini-Subrammanian** [comment]
@Subhasri-V-P  Is Unit test not necessary?

**Comment by Subhasri-V-P** [comment]
> @Subhasri-V-P Is Unit test not necessary?

The unit tests were not required to be done for nuget and analyzers assignment. As the Fibonacci generator function used is from a nuget package already tested

File: src/MyConsoleApplication/FibonacciGenerator.cs (hunk 1 above)
Comment (Subhasri-V-P) [reply src/MyConsoleApplication/FibonacciGenerator.cs:55]: Update code with ResetColor function.

File: src/MyConsoleApplication/FibonacciGenerator.cs (hunk 2 above)
Comment (Subhasri-V-P) [reply src/MyConsoleApplication/FibonacciGenerator.cs:42]: Updated code with reset color function.

**Comment by Subhasri-V-P** [comment]
Approved by mentor

**Comment by Subhasri-V-P** [comment]
Approved by mentor.

**Pull request closed without merging**